src/hash_map.c

```diff
@@ -230,6 +230,21 @@ static rcutils_ret_t hash_map_check_and_grow_map(rcutils_hash_map_t * hash_map)
   return ret;
 }
 
+// Modified from http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+static size_t next_power_of_two(size_t v)
+{
+  v--;
+  size_t shf = 1;
+  v |= v >> 1;
+  for (size_t i = 0; i < sizeof(size_t); ++i) {
+    shf <<= 1;
+    v |= v >> shf;
+  }
+  v++;
```

Comment:
how about the following?

```suggestion
  size_t shf;
  v--;
  for (size_t i = 0; i <= sizeof(size_t); ++i) {
    shf = (1 << i);
    v |= v >> shf;
  }
  v++;
```

Reply:
Good suggestion; looks like it is slightly more efficient, too.